torch/csrc/autograd/python_cpp_function_attr.cpp

```diff
@@ -0,0 +1,54 @@
+#include "python_cpp_function_attr.h"
+
+#include "torch/csrc/autograd/python_variable.h"
+#include "torch/csrc/autograd/python_hook.h"
+
+namespace torch { namespace autograd {
+
+namespace attributes {
+
+PyObject* next_functions(THPCppFunction* self, PyObject* hook)
```

Comment:
For convolution_attr.h: I wanted to avoid polluting convolution.h and convolution.cpp with Python API stuff. Are you thinking about having these definitions in init.cpp directly?